### PR TITLE
Setting intial theme using user preference

### LINKED ISSUE #6: Set light/dark theme based on user pref
When the theme is not explicitly set, use the [prefers-colors-scheme](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme) media query to set the right values

### PR DESCRIPTION
**Pull request by JayaKrishnaNamburu**
fixes #6

### CODE REVIEW

File: sandpack-react/src/contexts/theme-context.tsx [hunk 1]
```diff
@@ -19,13 +22,16 @@ const SandpackThemeContext = React.createContext<{
...
-  const { theme, id } = createThemeObject(props.theme);
+  let themeObj = createThemeObject(
+    getDarkModePreference() ? 'codesandbox-dark' : 'codesandbox-light'
+  );
   const c = useClasser('sp');
-
   // If theme is not explicitly set, don't inject any stylesheet
   if (props.theme) {
-    injectThemeStyleSheet(theme, id);
+    themeObj = createThemeObject(props.theme);
```
Comment (alexnm) [inline sandpack-react/src/contexts/theme-context.tsx:31]: if you make the changes above, you can keep the injectThemeStyleSheet behind the props.theme check

**Review by alexnm — CHANGES_REQUESTED** [review]

**Comment by JayaKrishnaNamburu** [comment]
sure alex, wll be making those changes 👍

File: sandpack-react/src/themes/index.ts [hunk 2]
```diff
@@ -175,9 +176,18 @@ export const SANDPACK_THEMES: Record<SandpackPredefinedTheme, SandpackTheme> = {
   'monokai-pro': monokaiProTheme,
 };
 
-export const createThemeObject = (
-  inputTheme?: SandpackPredefinedTheme | SandpackPartialTheme
-) => {
+export const createThemeObject = (inputTheme?: SandpackThemeProp) => {
+  if (inputTheme === 'auto') {
+    const theme = getDarkModePreference()
```
Comment (alexnm) [inline sandpack-react/src/themes/index.ts:181]: here you can also assign the id to `codesandbox-light` or `codesandbox-dark` so you don't generate a new hash for the theme. `auto` will still set a standard theme, the hash function is created for custom themes. Also, move this after the `if(inputTheme === undefined)`

**Review by alexnm — CHANGES_REQUESTED** [review]
Looks great, thanks for the effort! Just a small change request and we're good to go

**Review by alexnm — APPROVED** [review]

**Pull request merged**